libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -642,9 +640,36 @@ void CodeTransform::operator()(ForLoop const& _forLoop)
 	m_assembly.appendLabel(loopEnd);
 
 	finalizeBlock(_forLoop.pre, stackStartHeight);
+	m_context->forLoopStack.pop();
 	m_scope = originalScope;
 }
 
+int CodeTransform::adjustStackTo(int _targetDepth)
+{
+	int const stackDiffAfter = m_assembly.stackHeight() - _targetDepth;
+	for (int i = 0; i < stackDiffAfter; ++i)
+		m_assembly.appendInstruction(solidity::Instruction::POP);
+	return stackDiffAfter;
+}
+
+void CodeTransform::operator()(Break const& _break)
+{
+	yulAssert(!m_context->forLoopStack.empty(), "Invalid break-statement. Requires surrounding for-loop in code generation.");
+	m_assembly.setSourceLocation(_break.location);
+	auto const& labelInfo = m_context->forLoopStack.top().done;
+	m_assembly.appendJumpTo(labelInfo.label, adjustStackTo(labelInfo.stackHeightAdjust));
```

Comment:
if you call it like this, it sounds like `adjustStackTo` would not have side-effects.